sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/SupportsDelete.java

```diff
@@ -28,6 +28,25 @@
  */
 @Evolving
 public interface SupportsDelete {
+
+  /**
+   * Checks whether it is possible to delete data from a data source table that matches filter
+   * expressions.
+   * <p>
+   * Rows should be deleted from the data source iff all of the filter expressions match.
+   * That is, the expressions must be interpreted as a set of filters that are ANDed together.
+   * <p>
+   * Spark will call this method to check if the delete is possible without significant effort.
+   * Otherwise, Spark will try to rewrite the delete operation and produce row-level changes
+   * if the data source table supports deleting individual records.
+   *
+   * @param filters filter expressions, used to select rows to delete when all expressions match
+   * @return true if the delete operation can be performed
+   */
+  default boolean canDeleteWhere(Filter[] filters) {
+    return true;
```

Comment:
Shall we have `false` as a safer default?

Reply:
I'm wondering if there is a breaking change which we should have `true` here?

Reply:
Unfortunately, this would change the assumptions for existing implementations. Right now, if this interface is implemented, Spark will call `deleteWhere` for the delete. Returning `false` would cause Spark to skip it where the return of this method is used.

The original idea was to try to delete using `deleteWhere`, and if that fails to run a more expensive delete. But when we started implementing the more expensive delete, we needed to know during job planning, not job execution, whether the metadata-only delete can be done. This method solves that problem.

Reply:
Got it, @rdblue .

Reply:
That's correct and the method returns `true` to keep the old behavior by default.